src/main/scala/nl/knaw/dans/easy/multideposit/actions/CreateOutputDepositDir.scala

```diff
@@ -28,12 +28,28 @@ case class CreateOutputDepositDir(row: Int, datasetID: DatasetID)(implicit setti
   private val dirs = depositDir :: bagDir :: metadataDir :: Nil
 
   override def checkPreconditions: Try[Unit] = {
+    for {
+      _ <- checkDatasetIdIsValid
+      _ <- checkDirectoriesDoNotExist
```

Comment:
Not sure what you're talking about. You mean [this kind of co/contravariance](https://en.wikipedia.org/wiki/Covariance_and_contravariance_(computer_science))? More explanation please!